app/config/collections.php

```diff
@@ -1366,7 +1377,495 @@
+            ],
+            [
+                '$id' => ID::custom('_key_targetInternalId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['targetInternalId'],
+                'lengths' => [],
+                'orders' => [],
+            ],
+            [
+                '$id' => ID::custom('_key_topicId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['topicId'],
+                'lengths' => [],
+                'orders' => [],
+            ],
+            [
+                '$id' => ID::custom('_key_topicInternalId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['topicInternalId'],
+                'lengths' => [],
+                'orders' => [],
+            ]
+        ],
+    ],
+
+    'targets' => [
+        '$collection' => ID::custom(DATABASE::METADATA),
+        '$id' => ID::custom('targets'),
+        'name' => 'Targets',
+        'attributes' => [
+            [
+                '$id' => ID::custom('userId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('userInternalId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('providerId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('providerInternalId'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => [],
+            ],
+            [
+                '$id' => ID::custom('providerType'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => false,
+                'default' => null,
+                'array' => false,
+                'filters' => ['subQueryProviderType'],
+            ],
```

Comment:
Do we actually need this 🤔 might be just as easy to get the provider and check it's `type` after

Reply:
If it is possible we prefer types as constant integers.

Reply:
hmm, we can change this to provider as well. although we are only selecting type from provider sub query so that would still be faster than fetching the whole provider doc but your call @abnegate

Reply:
Let's remove it, and call `getDocument` using the `providerId` from here where we need it. Then if we need any other provider attributes we can use them, without making a second query to get the same document but with more attributes